packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -1,18 +1,92 @@
-import cloudformation = require('@aws-cdk/aws-cloudformation');
-import cloudfront = require('@aws-cdk/aws-cloudfront');
-import iam = require('@aws-cdk/aws-iam');
-import lambda = require('@aws-cdk/aws-lambda');
-import s3 = require('@aws-cdk/aws-s3');
-import cdk = require('@aws-cdk/core');
-import { Token } from '@aws-cdk/core';
+import cloudformation = require("@aws-cdk/aws-cloudformation");
+import cloudfront = require("@aws-cdk/aws-cloudfront");
+import iam = require("@aws-cdk/aws-iam");
+import lambda = require("@aws-cdk/aws-lambda");
+import s3 = require("@aws-cdk/aws-s3");
+import cdk = require("@aws-cdk/core");
+import { Token } from "@aws-cdk/core";
 import crypto = require('crypto');
+import path = require("path");
 import fs = require('fs');
-import path = require('path');
-import { ISource, SourceConfig } from './source';
+import { ISource, SourceConfig } from "./source";
 
-const handlerCodeBundle = path.join(__dirname, '..', 'lambda', 'bundle.zip');
+const handlerCodeBundle = path.join(__dirname, "..", "lambda", "bundle.zip");
 const handlerSourceDirectory = path.join(__dirname, '..', 'lambda', 'src');
 
+export type CacheControlDirective =
+  | "must-revalidate"
+  | "no-cache"
+  | "no-store"
+  | "no-transform"
+  | "public"
+  | "private"
+  | "proxy-revalidate"
+  | { "max-age": cdk.Duration }
+  | { "s-max-age": cdk.Duration }
+  | string;
+export interface SystemDefinedObjectsMetadata {
+  readonly "cache-control"?: CacheControlDirective[];
```

Comment:
- Use Jsii-compatible key: cacheControl
- you will need to model cacheControl as a “union-like” class:

```ts
export class CacheControl {
  public static mustRevalidate() { return new CacheControl(“must-recalibrate”); }
  public static maxAge(t: Duration) { return new CacheControl({ “max-age”: t.seconds() }); }
  // ...

  private constructor(public value: any) {}
}
```